x-pack/legacy/plugins/uptime/server/lib/adapters/monitor_states/search/__tests__/query_context.test.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { QueryContext } from '../query_context';
+import { CursorPagination } from '../..';
+import { CursorDirection, SortOrder } from '../../../../../../common/graphql/types';
+
+describe(QueryContext, () => {
+  // 10 minute range
+  const rangeStart = '2019-02-03T19:06:54.939Z';
+  const rangeEnd = '2019-02-03T19:16:54.939Z';
+
+  const pagination: CursorPagination = {
+    cursorDirection: CursorDirection.AFTER,
+    sortOrder: SortOrder.DESC,
+  };
+
+  let qc: QueryContext;
+  const makeQueryContext = () => {
+    return new QueryContext({}, rangeStart, rangeEnd, pagination, null, 10);
+  };
+  beforeEach(() => (qc = makeQueryContext()));
+
+  describe('dateRangeFilter()', () => {
+    const expectedRange = {
+      range: {
+        '@timestamp': {
+          gte: rangeStart,
+          lte: rangeEnd,
+        },
+      },
+    };
+    describe('when hasTimespan() is true', () => {
+      it('should create a date range filter including the timespan', async () => {
+        const mockHasTimespan = jest.fn();
+        mockHasTimespan.mockReturnValue(Promise.resolve(true));
```

Comment:
```suggestion
        mockHasTimespan.mockReturnValue(true);
```
I don't think we need the `Promise.resolve`.

Reply:
Are you sure? That function is `async`, so should always return a promise.